client/src/App.jsx

```diff
@@ -189,27 +200,31 @@ class App extends React.Component {
   }
 
   changeHoveredDate(date) {
-    var hDate = new Date(date);
-    hDate.setHours(0, 0, 0);
+    var hDate = availabilityHelpers.getDateObjFromStr(date);
     this.setState({
       hoveredDate: hDate.toString()
     });
   }
 
-  getTotalPrice(checkOut) {
+  getTotalPrice(checkOut, checkIn, dates) {
```

Comment:
Just thinking about function names and the scope of each function. Most of your other `getXYZ()` functions are returning a value to be used in another function. While this function does get pricing information, it also makes many state changes. Does the name match the result? Maybe it could be refactored later for clarity.